File: src/Billing.Mediation/Sql/ISqlExecutor.cs
namespace Billing.Mediation.Sql;

/// <summary>
/// Runs one SQL statement on the single shared MySqlConnection (and its transaction) and returns the
/// affected-row count — the thin replacement for the legacy <c>DbCommand</c> + <c>DbWriterWithAccurateCount</c>
/// stored proc. Batching is layered on top by <see cref="BatchSqlWriter"/> (the ported
/// <c>CollectionSegmenter</c>). The live implementation wraps one <c>MySqlCommand</c> on the per-call
/// connection; tests use an in-memory capture.
/// </summary>
public interface ISqlExecutor
{
    int ExecuteNonQuery(string sql);
}

File: java/src/main/java/com/telcobright/billing/mediation/sql/ISqlExecutor.java
package com.telcobright.billing.mediation.sql;

/**
 * Runs one SQL statement on the single shared MySqlConnection (and its transaction) and returns the
 * affected-row count — the thin replacement for the legacy {@code DbCommand} + {@code DbWriterWithAccurateCount}
 * stored proc. Batching is layered on top by {@link BatchSqlWriter} (the ported
 * {@code CollectionSegmenter}). The live implementation wraps one {@code MySqlCommand} on the per-call
 * connection; tests use an in-memory capture.
 */
public interface ISqlExecutor {
    int ExecuteNonQuery(String sql);
}
